src/ebay-textbox/textbox.tsx

```diff
@@ -1,35 +1,59 @@
-import React, { ChangeEvent, cloneElement, ComponentProps, FC, FocusEvent, Ref, useEffect, useState } from 'react'
+import React, {
+    cloneElement, ComponentProps, FC, Ref,
+    ChangeEvent, FocusEvent, KeyboardEvent, MouseEvent, SyntheticEvent,
+    useEffect, useState
+} from 'react'
 import classNames from 'classnames'
 import { findComponent, withForwardRef } from '../common/component-utils'
 import { EbayTextboxPostfixIcon, EbayTextboxPrefixIcon, Size } from './index'
 import { useFloatingLabel } from '../common/floating-label-utils/hooks'
+import {
+    EbayChangeEventHandler,
+    EbayEventHandler,
+    EbayFocusEventHandler,
+    EbayKeyboardEventHandler, EbayMouseEventHandler
+} from '../common/event-utils/types'
 
 const isControlled = value => typeof value !== 'undefined'
 
 type TextInputProps = ComponentProps<'input'> & ComponentProps<'textarea'>
 
+export type TextboxEventProps = { value: string }
 export type EbayTextboxProps = {
     fluid?: boolean;
     invalid?: boolean;
     multiline?: boolean;
     defaultValue?: string;
     inputSize?: Size;
-    onFocus?: (e?: FocusEvent<HTMLTextAreaElement & HTMLInputElement>, value?: string) => void;
-    onBlur?: (e?: FocusEvent<HTMLTextAreaElement & HTMLInputElement>, value?: string) => void;
-    onChange?: (e: ChangeEvent<HTMLTextAreaElement & HTMLInputElement>, value: string) => void;
-    onButtonClick?: (e: MouseEvent, value: string) => void;
     floatingLabel?: string;
+    onChange?: EbayChangeEventHandler<HTMLTextAreaElement & HTMLInputElement, TextboxEventProps>;
+    onInputChange?: EbayChangeEventHandler<HTMLTextAreaElement & HTMLInputElement, TextboxEventProps>;
```

Comment:
Why are we introducing onInputChange since we have onChange already?

Reply:
`onChange` is triggered when you leave the field
`onInputChange` is triggered on any value change

for compatibility reasons with marko ebayUI:
https://opensource.ebay.com/ebayui-core/?path=/story/form-input-ebay-textbox--with-label
<img width="491" alt="image" src="https://github.com/eBay/ebayui-core-react/assets/2737310/b8cd7a57-647e-4c6a-8091-6ff70fd3d5cf">